Wbooks/src/app/screens/library/components/bookList/index.js

```diff
@@ -1,27 +1,60 @@
 import React, { Component } from 'react';
-import { FlatList } from 'react-native';
+import { FlatList, View } from 'react-native';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 import Book from '../book/index';
-import mockData from '../../../../mocklist.json';
+import BookActions from '../../../../../redux/book/actions';
+import bookModel from '../../../../proptypes/bookModel';
 
 import styles from './styles';
 
 class BookListContainer extends Component {
-  renderItem = ({ item }) => <Book book={item} navigation={this.props.navigation} />;
+  handlePress = item => {
+    const { navigation, loadBook } = this.props;
+    loadBook(item);
+    navigation.navigate('BookDetail', { navigation });
+  };
+
+  renderItem = ({ item }) => <Book book={item} onPress={this.handlePress} />;
 
   keyExtractor = item => item.id.toString();
 
+  componentDidMount() {
+    this.props.getBookList(BookActions.getBookList());
```

Comment:
`this.props.getBookList` = `() => dispatch(BookActions.getBookList())`, it doesn't expect any argument and is already dispatching an action.
`this.props.getBookList();` should be enough